src/core/Akka.Remote/Endpoint.cs

```diff
@@ -1842,10 +1842,57 @@ public OutboundAck(Ack ack)
 
     }
 
+    internal sealed class MsgDispatcherActor : ReceiveActor
```

Comment:
Since this is a hot path and we don't need `ReceiveAsync` behavior here, should this be a more base actor to save some delegate calls?

Reply:
Done - measuring this change now...

Reply:
Performance appears to be noticeably worse - 20k msg/s worse with `UntypedActor` and an `if... is` statement than a `ReceiveActor`.

Been able to reproduce it in both directions several times - going to revert probably.

Reply:
Done - reverted it. I think it was @ismaelhamed who pointed out that the `UntypedActor` itself may be faster, but the way `Receive<T>` works in the receive actor is inherently faster than some of what happens when powering `switch` and `if .. is _` statements. For whatever reason, that appears to be what's at work here.